cli/internal/core/engine.go

```diff
@@ -446,13 +446,16 @@ func (e *Engine) getTaskDefinitionChain(taskID string, taskName string) ([]fs.Bo
 				)
 			}
 
-			// TODO(mehulkar):
-			// 		Pipeline.GetTask allows searching with a taskID (e.g. `package#task`).
-			// 		But we do not want to allow this, except if we're in the root workspace.
-			workspaceDefinition, err := workspaceTurboJSON.Pipeline.GetTask(taskID, taskName)
-
-			if err == nil {
-				taskDefinitions = append(taskDefinitions, *workspaceDefinition)
+			if workspaceDefinition, ok := workspaceTurboJSON.Pipeline[taskName]; ok {
+				taskDefinitions = append(taskDefinitions, workspaceDefinition)
+			} else {
+				// If the workspace doesn't have the taskName defined, but we detect that
+				// it defines the package#task syntax, throw an error. This is a misconfiguration
+				// and we actively want to prevent it.
+				// TODO(mehulkar): Should we validate this upfront when reading a worksapce turbo.jsoN?
```

Comment:
Probably yes. Would it help to have `GetTurboConfigFromWorkspace` return `nil, nil` to signal not found to free up the error for actual errors that should be reported? Another option that is somewhat common is to check if the returned error `os.IsNotExist(err)`. IMHO that puts additional work on the caller, whereas checking that a returned value is `nil` or not is very slightly more obvious to do. Another option is that I've seen would be to change the return type to `(*TurboJSON, bool, error)` and treat the `bool` as found / not found. The plus side to this approach is that it mimics checking a map for a key.